enhancements/olm/platform-operators.md

```diff
@@ -0,0 +1,435 @@
+
+In addition, future phase requirements around package dependency resolution may require a new design entirely. While this is a notable short term trade-off, it allows the OLM team to continue iterating on their "deppy" component in the background. Deppy is an OLM 1.x component that's responsible for dependency resolution. A future design may elect to use deppy and it's APIs, and delegate package selection to that component. The usage of feature flags can be used to toggle the underlying implementation layer once that component is ready.
+
+This phase will likely restrict which OLM-based operators admins can install. Those requirements will be detailed below. The immediate pool of candidates may include the cert-manager and service-binding-operator operators.
+
+#### Phase 1: Expand Capabilities
+
+Summary: Introduce dependency resolution, and expand the set of supported capabilities for admins and platform operator authors. Help establish, or improve, any first-class APIs at the installer and cluster runtime levels.
+
+The previous phase aimed to get the foundational layer in place to start onboarding more OLM-based operators to participate in the cluster's lifecycle. The notable limitations with phase 0 is requiring manual admin intervention around editing the installer's Kubernetes manifests and upgrading PO packages after installation. This phase will aim to address those limitations by introducing first-class concepts.
+
+This phase will also attempt to introduce package management and dependency resolution primitives. Candidate PO packages may place dependencies on other packages and GVKs, in addition to more granular constraints that are needed in order to accommodate EUS-EUS drift and platform-specific cluster invariants.
+
+#### Phase 2: Quality of Life (QoL) Improvements
+
+Summary: Prepare for general availability by tackling any performance improvements, shoring up the overall design/implementation/UX, and any QoL improvements that have been identified from previous phase feedback.
+
+This phase has largely been unexplored. Candidates for this phase may include:
+
+- Integration with the OCP console, hypershift, KCP, etc.
+- Improved instrumentation.
+- Improved performance of the overall stack.
+- Improved UX around the user-facing APIs.
+
+### PO Requirements
+
+- Must package Kubernetes manifests using OLM's registry+v1 bundle format.
+- Cannot specify non-AllNamespace CSV install modes.
+- Cannot declare package or GVK dependencies.
+- All package bundles must live in the redhat-operators CatalogSource.
+
+### High-Level Design
+
+In order to enable admins to express their desired intent to install an OLM-based operator in day 0 operations, a new, top-level `PlatformOperator` cluster-scoped CRD will be introduced. This resource can help serve as the bridge between the openshift installer, and the on-cluster components.
```

Comment:
Can't I already express that intent today by including a manifest of OLM resources at install time? What value does the new CRD bring?

Reply:
it brings CVO awareness (as proxied through the platform operator controller), meaning that we can ensure the install+upgrades are not considered complete until these POs are installed/upgraded, which is not something we are going to attempt to provide for operators installed via the standard OLM install process.

it also eventually will allow us to create a first class UX around platform operators that is more similar to what we do for CVO managed operators (e.g. indications when they are not available, are degraded, etc) than what is provided for OLM operators.

Reply:
Makes sense. I'd suggest replacing the "In order to..." sentence with the points you describe.

Reply:
> Can't I already express that intent today by including a manifest of OLM resources at install time? What value does the new CRD bring?

Yep, this is a good call out. I updated the language in the recent set of commits, and massaged some of Ben's points into the proposal. Hopefully that updated language is a bit clearer now.